src/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -847,19 +847,20 @@ void QgsGdalProvider::readBlock( int bandNo, QgsRectangle  const &extent, int pi
   QgsDebugMsgLevel( QStringLiteral( "tmpXMin = %1 tmpYMax = %2 tmpWidth = %3 tmpHeight = %4" ).arg( tmpXMin ).arg( tmpYMax ).arg( tmpWidth ).arg( tmpHeight ), 5 );
 
   // Allocate temporary block
-  char *tmpBlock = ( char * )qgsMalloc( dataSize * tmpWidth * tmpHeight );
+  size_t bufferSize = dataSize * static_cast<size_t>( tmpWidth ) * static_cast<size_t>( tmpHeight );
```

Comment:
on 32 bit builds, that might overflow, so if we want to be bullet proof, that computation should be done explicitly on quint64 (or whatever QT has for a uint64), cast to size_t, compare the 2 values, and error out if they are different

Reply:
@rouault I'm not sure I got that right, can you please have a look to https://github.com/qgis/QGIS/pull/8550/commits/b3ed482bc8f2ed1408d9ae1e834d53fff771edc9#diff-e743a574c60fb6098a9e6f631e9aadabR851 ?